ledger/sync/src/reqwest_transactions_fetcher.rs

```diff
@@ -164,34 +174,34 @@ impl ReqwestTransactionsFetcher {
 
         let s3_block_data = S3BlockData {
```

Comment:
There should probably be a check here that verifies that block.contents_hash == block_contents.hash()